pkg/types/powervs/validation/platform.go

```diff
@@ -44,6 +49,68 @@ func ValidatePlatform(p *powervs.Platform, fldPath *field.Path) field.ErrorList
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("ServiceInstanceGUID"), p.ServiceInstanceGUID, "ServiceInstanceGUID must be a valid UUID"))
 		}
 	}
+	if p.ServiceEndpoints != nil {
+		allErrs = append(allErrs, validateServiceEndpoints(p.ServiceEndpoints, fldPath.Child("serviceEndpoints"))...)
+	}
 
 	return allErrs
 }
+
+// validateServiceEndpoints checks that the specified ServiceEndpoints are valid.
+func validateServiceEndpoints(endpoints []configv1.PowerVSServiceEndpoint, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	tracker := map[string]int{}
+	for index, endpoint := range endpoints {
+		fldp := fldPath.Index(index)
+		if eindex, ok := tracker[endpoint.Name]; ok {
+			allErrs = append(allErrs, field.Invalid(fldp.Child("name"), endpoint.Name, fmt.Sprintf("duplicate service endpoint not allowed for %s, service endpoint already defined at %s", endpoint.Name, fldPath.Index(eindex))))
+		} else {
+			tracker[endpoint.Name] = index
+		}
```

Comment:
Any reason not to use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets? For me it communicates the intention faster:
```suggestion
	knownEndpoints := sets.New[string]()
	for index, endpoint := range endpoints {
		fldp := fldPath.Index(index)
		if knownEndpoints.Has(endpoint.Name) {
			allErrs = append(allErrs, field.Duplicate(fldp.Child("name"), endpoint.Name))
		} 
		knownEndpoints.Add(endpoint.Name)
```

Reply:
I can do that! Full disclosure this function is copy pasted from the `ibmcloud` provider. Initially I tried to import and reuse it but the endpoint types are different.